### ISSUE TITLE
Clip rotation does not work for RGBA images in 90 degrees steps

### ISSUE THREAD

**Issue by GrigoreAlexandru** [Zulko/moviepy#1126]
### Expected Behavior
clip rotates

### Actual Behavior
ValueError: axes don't match array

### Steps to Reproduce the Problem

```
img = PIL.Image.new("RGBA",(10,10))
array = np.array(img)
clip = ImageClip(array).add_mask().rotate(90)
```

### Specifications

  - Python Version: 3.7
  - Moviepy Version: 1.0.2
  - Platform Name: Windows
  - Platform Version: 10

### More info

from rotate.py

```
transpo = [1,0] if clip.ismask else [1,0,2]

....        
im = gf(t)
....
        if (a==90) and expand:
            return np.transpose(im, axes=transpo)[::-1]
```
This happens because gf() returns the frame 2 times, once with the RGB matrix and the other one with the alpha channel matrix.

So to fix this you would need a way to find out what matrix you're getting and pass the proper axes to transpose. This is done with the
```
transpo = [1,0] if clip.ismask else [1,0,2]
```
but it's weird since the clip is constant and so is .ismask, furthermore, from my findings, ismask returns false even when using clip.add_mask(), not sure what gives.

One other way is to use rot90 from numpy instead of transpose which eliminates the need to differentiate between the matrices.

**Comment by ODtian** [comment]
hi, I use fl() method to make a opacity animation, but I got an error occured, I use gf func to get numpy array and use pillow to set a opacity to it then convert to nparray to return, is there any thing need to be noticed for the rgba array?

**Comment by GrigoreAlexandru** [comment]
> hi, I use fl() method to make a opacity animation, but I got an error occured, I use gf func to get numpy array and use pillow to set a opacity to it then convert to nparray to return, is there any thing need to be noticed for the rgba array?

What error are you getting ?
Debug step by step whatever gf() returns, check the matrix size on each step. Most likely you're getting first a RGB matrix then on the second return you get the alpha matrix. 2 solutions on top of my head are to either merge the 2 matrices and use pillow to modify the alpha or just work with the alpha matrix alone.

You can also animate clips by returning a closure to set_opacity().

**Comment by tburrows13** [comment]
Fixed in #1335
